### PR TITLE
PKX-673 Added email support

### PR DESCRIPTION
**Pull request by m-usman01**
### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: common/djangoapps/util/milestones_helpers.py [hunk 1]
```diff
@@ -373,8 +379,10 @@ def get_course_content_milestones(course_id, content_id=None, relationship='requ
         )
 
     if content_id is None:
+        log.info("content id none returning:{}".format(request_cache_dict[user_id][relationship]))
         return request_cache_dict[user_id][relationship]
 
+    log.info("returning")
```
Comment (emzubair) [inline common/djangoapps/util/milestones_helpers.py:385]: we can remove this line?

File: lms/djangoapps/course_api/blocks/transformers/milestones.py [hunk 2]
```diff
@@ -79,9 +79,12 @@ def user_gated_from_block(block_key):
             return False
 
         for block_key in block_structure.topological_traversal():
+            log.info("\n\nChecking for {} include gated:{}".format(block_key, self.include_gated_sections))
             if user_gated_from_block(block_key):
+                log.info("user gated from block")
                 block_structure.remove_block(block_key, False)
             elif self.include_gated_sections and self.has_pending_milestones_for_user(block_key, usage_info):
+                log.info("add gated flag")
```
Comment (emzubair) [inline lms/djangoapps/course_api/blocks/transformers/milestones.py:87]: please refactor these logs

**Review by emzubair — CHANGES_REQUESTED** [review]

File: openedx/features/pakx/cms/custom_settings/views.py [hunk 3]
```diff
@@ -119,25 +120,45 @@ def get_chapters_and_subsections(self, course, request):
...
         """Update course milestones if subsection to lock has been changed."""
 
         course_overview_content = CourseOverviewContent.objects.get(course_id=course_key)
 
         if subsection and course_overview_content.subsection_to_lock != subsection:
-            self.update_course_milestone(course_overview_content.subsection_to_lock, subsection, course_key)
+            self._update_course_milestone(course_overview_content.subsection_to_lock, subsection, course_key)
 
-    def update_course_milestone(self, old_subsection, new_subsection, course_key):
+    def _update_course_milestone(self, old_subsection: str, new_subsection: str, course_key: str):
```
Comment (emzubair) [inline openedx/features/pakx/cms/custom_settings/views.py:152]: would suggest to remove type annotations for symmetry

**Review by emzubair — DISMISSED** [review]

**Review by mubtadaali — APPROVED** [review]
LGTM 👍

**Pull request merged**